scripts/library.sh

```diff
@@ -415,7 +418,20 @@ function is_int() {
 # Return whether the given parameter is the knative release/nightly GCF.
 # Parameters: $1 - full GCR name, e.g. gcr.io/knative-foo-bar
 function is_protected_gcr() {
-  [[ -n $1 && "$1" =~ "^gcr.io/knative-(releases|nightly)/?$" ]]
+  [[ -n $1 && $1 =~ ^gcr.io/knative-(releases|nightly)/?$ ]]
```

Comment:
Are the quotes giving trouble?

Reply:
not here, the one below given me trouble:
`[[ "gke_knative-tests_us-central1-f_prow" =~ "^gke_knative-tests_us\-[a-zA-Z0-9]+\-[a-z]+_prow$" ]] &
& echo "match" || echo "no match"` gave me `no match`, and removing `"` got `match`

change it here for consistency

Reply:
I think it's the backslash. Can we get unit tests for the new/changed methods please? :)

Reply:
actually I just tested, `[[ "gcr.io/knative-releases" =~ "^gcr.io/knative-(releases|nightly)/?$" ]] && echo "match" || echo "no match"` gave me "no match` as well

Reply:
agree with unit test, will do

Reply:
unit tests added, and decided to extend protection on all clusters under `knative-tests`